index.ts

```diff
@@ -0,0 +1,20 @@
+import { System, SystemCommand } from './types/SystemTypes.ts'
+import { activityMonitor } from './events/activityMonitor.ts'
+import { logActivity } from './events/logActivity.ts'
+import os from 'os';
+import fs from 'fs';
+
+const systemType = os.type();
+const liveActivity = activityMonitor(SystemCommand[systemType as System])
```

Comment:
This in fact is not guaranteed, because there are many more systems that Node.js recognizes, not just three. In case the system was something else `SystemCommand[systemType as System]` would return undefined and crash the app ( I know you are trying to handle errors, but I believe those errors are the ones you get from the os for not being able to perform an action - lack of permission for example). The input to the `child_process.exec()` function should be a string, cannot be undefined afaik...

Uncaught:
TypeError [ERR_INVALID_ARG_TYPE]: The "command" argument must be of type string. Received undefined
    at validateString (node:internal/validators:162:11)
    at normalizeExecArgs (node:child_process:190:3)
    at exec (node:child_process:233:16) {
  code: 'ERR_INVALID_ARG_TYPE'
}

Reply:
As memntioned in class: It's enough to filter out win32 and assume the rest is UNIX based.